modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchSearchEngine.java

```diff
@@ -285,19 +288,48 @@ public void setAutoCreateIndex(boolean enable) {
 		ClusterUpdateSettingsRequest clusterUpdateSettingsRequest =
 			new ClusterUpdateSettingsRequest();
 
-		clusterUpdateSettingsRequest.persistentSettings(
-			Settings.builder(
-			).put(
-				"action.auto_create_index",
-				_createAutoCreateIndexSetting(enable)
-			));
-
 		try {
+			clusterUpdateSettingsRequest.persistentSettings(
+				Settings.builder(
+				).put(
+					"action.auto_create_index",
+					_createAutoCreateIndexSetting(enable)
+				));
+
 			clusterClient.putSettings(
 				clusterUpdateSettingsRequest, RequestOptions.DEFAULT);
 		}
+		catch (ElasticsearchStatusException elasticsearchStatusException) {
+			if (Objects.equals(
+					elasticsearchStatusException.status(),
+					RestStatus.FORBIDDEN) ||
+				Objects.equals(
+					elasticsearchStatusException.status(),
+					RestStatus.UNAUTHORIZED)) {
+
+				StringBundler sb = new StringBundler(4);
+
+				sb.append("Cluster auto create index setting could not be ");
+				sb.append("updated due to lacking permissions. This can lead ");
+				sb.append("to index mappings being created incorrectly: ");
+
+				if (_log.isDebugEnabled()) {
+					sb.append(elasticsearchStatusException);
+
+					_log.debug(sb.toString());
+				}
+				else if (_log.isWarnEnabled()) {
```

Comment:
the default level here is going to be ERROR, so in the case of permission denied, i dont think anything will actually be logged, unless the user had set different log levels themselves... @lipusz thoughts on this? if you want a message to be displayed by default, we could bump the default level for this class to WARN, or change the logging to ERROR

Reply:
Yes, by default it's ERROR and nothing logged. Maybe i misread you comment [here](https://liferay.atlassian.net/browse/PTR-5609?focusedCommentId=2759341) but I thought using WARN was your idea?

Reply:
WARN sounds good to me. It doesn't break the whole index creation, rt? So ERROR would be not appropriate as far as I understand.